# U.S. Medical Insurance Costs

In this project, we are going to analize a dataset about the U.S. Medical Insurance Costs. The goal of the project is using the Python fundamentals I learnt about in the first two months of 2021 and extrapolate as much data as possible.
The Python fundamentals I learnt in this carrer path have been:
* Syntax
* Variables
* Functions
* Control Flow 
* Lists
* Loops
* Strings
* Dictionaries
* Classes
* Modules
* Files

In [2]:
import csv

First thing first, I import the csv library, since the file we are going to use is in a .csv form.

In [90]:
with open('insurance.csv') as insurance:
    csv_info = insurance.readline()
    csv_categories = csv_info.split(",")
    age_key = csv_categories[0]
    sex_key = csv_categories[1]
    bmi_key = csv_categories[2]
    children_key = csv_categories[3]
    smoker_key = csv_categories[4]
    region_key = csv_categories[5]
    charges_key = "charges"

    age_value = []
    sex_value = []
    bmi_value = []
    children_value = []
    smoker_value = []
    region_value = []
    charges_value = []
    
    csv_data = insurance.readlines()
    for line in csv_data:
        csv_category = line.split(",")
        age_value.append(csv_category[0])
        sex_value.append(csv_category[1])
        bmi_value.append(csv_category[2])
        children_value.append(csv_category[3])
        smoker_value.append(csv_category[4]) 
        region_value.append(csv_category[5])
        charges_value.append(csv_category[6][:-2])
    
    dataset = {}
    dataset[age_key] = age_value
    dataset[sex_key] = sex_value
    dataset[bmi_key] = bmi_value
    dataset[children_key] = children_value
    dataset[smoker_key] = smoker_value
    dataset[region_key] = region_value
    dataset[charges_key] = charges_value
    
    print(len(dataset['age']))

1338


(N.B. future me will hate these line of codes) with this line of code we create a usable way to use our dataset of 1338 patients, putting every category (age, sex, ...) as a key of a long list of data.
Now we can easily access every category of our reports:
* (0) Patient Age (numerical discrete)
* (1) Patient Sex (categorical nominal)
* (2) Patient Bmi (numerical continuous)
* (3) Patient Number of children (numerical discrete)
* (4) Patient Smoking status (categorical nominal)
* (5) Patient Region (categorical nominal)
* (6) Patient Charge (numerical continuous)

Let's start listing some of the analysis we want to do with this data:

- Is the average age of this dataset equal to the mean of the USA population (38,2)?
- Where do most people come from?
- What is the average difference in cost between smokers and non-smokers?
- How many of the patients are smokers?
- What is the average age of people without children? While the mean of people with children?
- What is the proportions of smokers between the two sexes?
- What is the average age per region?
- ...

In [18]:
sum_of_ages = 0
for age in dataset['age']:
    sum_of_ages += int(age)
average_age = sum_of_ages / len(dataset['age'])
print(average_age)

39.20702541106129


As we can see the average age in this dataset is greater than the mean of the USA population. One of the reasons of this discrepancy could be a higher chance of having a medical condition at an older age and/or the absence of minors in this dataset.

In [125]:
regions = {}
for region in dataset['region']:
    if region not in regions.keys():
        regions[region] = 1
    else:
        regions[region] = regions[region] + 1

for region in regions.keys():
    percent = (regions[region] / len(dataset['region'])) * 100
    print(region + " has " + str(percent) + "% of people in the dataset (" +str(regions[region]) + " patients)")

southwest has 24.28998505231689% of people in the dataset (325 patients)
southeast has 27.204783258594915% of people in the dataset (364 patients)
northwest has 24.28998505231689% of people in the dataset (325 patients)
northeast has 24.2152466367713% of people in the dataset (324 patients)


We can see how we have a somewhat equal percent of people that come from all regions, with an explicable high percent of patients from southeast.

In [89]:
average_cost_smoker = 0
average_cost_non_smoker = 0

denominator_smoker = 0
denominator_non_smoker = 0
for index in range(0, len(dataset['smoker'])):
    
    if dataset['smoker'][index] == 'yes':
        denominator_smoker += 1
        average_cost_smoker += float(dataset['charges'][index])
   
    elif dataset['smoker'][index] == "no":
        denominator_non_smoker += 1
        average_cost_non_smoker += float(dataset['charges'][index])

average_cost_smoker = (average_cost_smoker / denominator_smoker)
average_cost_non_smoker = (average_cost_non_smoker / denominator_non_smoker)
print('The average cost of a smoker is ' + str(average_cost_smoker) + "dollars.")
print('The average cost of a non smoker is ' + str(average_cost_non_smoker) + "dollars.")
print('The average difference between a smoker and a non smoker is ' + str((average_cost_smoker - average_cost_non_smoker)) + "dollars.")

percent = 0
for i in dataset['smoker']:
    if i == 'yes':
        percent += 1
    else:
        continue
percent = (percent / len(dataset['smoker'])) * 100
print("The percent of smokers is " + str(percent) + "%.")

The average cost of a smoker is 31923.07308978102dollars.
The average cost of a non smoker is 8434.264588637203dollars.
The average difference between a smoker and a non smoker is 23488.808501143816dollars.
The percent of smokers is 20.47832585949178%.


We answer two questions with the same code: first we can notice how being a smoker increases your bill by about 23488$. We can see that about 20 percent of patients are smokers, 5 percent more than the USA estimated percent (15 percent according to CDC), with this information we can clearly say that being a smoker raises your chances of having some medical conditions in the near future.

In [108]:
average_age_w_children = 0
average_age_wo_children = 0
list_w_children = 0
list_wo_children = 0
for index in range(0, len(dataset['children'])):
    num_of_children = dataset['children'][index]
    
    if num_of_children != '0':
        average_age_w_children += int(dataset['age'][index])
        list_w_children += 1
        
    else:
        average_age_wo_children += int(dataset['age'][index])
        list_wo_children += 1
average_age_w_children = average_age_w_children / list_w_children
average_age_wo_children = average_age_wo_children / list_wo_children
print("The average age of a patient with children is " + str(average_age_w_children))
print("The average age of a patient without children is " + str(average_age_wo_children))

The average age of a patient with children is 39.78010471204188
The average age of a patient without children is 38.444250871080136


As we can see there is a little difference between people with and without children, since the probability of someone being a parent increases as time goes on.

In [113]:
num_of_male_smokers = 0
num_of_female_smokers = 0
smokers = 0
for index in range(0, len(dataset['smoker'])):
    if dataset['smoker'][index] == 'yes':
        
        if dataset['sex'][index] == "male":
            num_of_male_smokers += 1
        else:
            num_of_female_smokers += 1
        smokers += 1
        
percent_male = (num_of_male_smokers / smokers) * 100
percent_female = (num_of_female_smokers / smokers) * 100
print("In this dataset there are " + str(num_of_male_smokers) + " male smokers (" + str(percent_male) + " percent of all smokers.)")
print("In this dataset there are " + str(num_of_female_smokers) + " female smokers (" + str(percent_female) + " percent of all smokers.)")

In this dataset there are 159 male smokers (58.02919708029197 percent of all smokers.)
In this dataset there are 115 female smokers (41.97080291970803 percent of all smokers.)


The data we just studied reveals how about 58% of the smokers are male.

In [134]:
average_age_per_region = {}
len_per_region = {}
for index in range(0, len(dataset["region"])):
    if dataset["region"][index] not in average_age_per_region.keys():
        average_age_per_region[dataset["region"][index]] = 0
        len_per_region[dataset["region"][index]] = 0
    else:
        average_age_per_region[dataset["region"][index]] += int(dataset["age"][index])
        len_per_region[dataset["region"][index]] += 1

for index in average_age_per_region.keys():
         average_age_per_region[index] = average_age_per_region[index] / len_per_region[index]
         print(index + " has an average age of " +str(average_age_per_region[index]))

southwest has an average age of 39.51851851851852
southeast has an average age of 38.99724517906336
northwest has an average age of 39.21604938271605
northeast has an average age of 39.27554179566563


No important differences can be seen by looking at this data.

This is the end of my project; in the future I may extend this data analysis with graphs and/or revisiting my code.
I'm really satisfied with how I approached this project and I hope it won't be the last!